mongorestore/restore.go

```diff
@@ -25,24 +25,25 @@ import (
 
 const insertBufferFactor = 16
 
-var indexOptions = map[string]bool{
-	"bits": true,
-	"min":  true,
-	"max":  true,
-	"coarsestIndexedLevel": true,
-	"finestIndexedLevel":   true,
-	"2dsphereIndexVersion": true,
-	"background":           true,
-	"collation":            true,
-	"default_language":     true,
-	"dropDups":             true,
-	"expireAfterSeconds":   true,
-	"bucketSize":           true,
-	"name":                 true,
-	"v":                    true,
-	"key":                  true,
-	"language_override":    true,
-	"ns":                   true,
+// validIndexOptions are taken from https://github.com/mongodb/mongo/blob/master/src/mongo/db/index/index_descriptor.h
+var validIndexOptions = map[string]bool{
+	"bits":                    true,
+	"min":                     true,
+	"max":                     true,
+	"coarsestIndexedLevel":    true,
+	"finestIndexedLevel":      true,
+	"2dsphereIndexVersion":    true,
+	"background":              true,
+	"collation":               true,
+	"default_language":        true,
+	"dropDups":                true,
```

Comment:
IIUC, `dropDups` isn't currently valid.  I wonder if you need to make this map version-specific as it may change over time. Or populate it with some version-specific data at startup.